Documentation/Relationship/Relationship.md

```diff
@@ -0,0 +1,334 @@
+The destination can of course also include `@PathParameters`.
+
+Any manual definition of a relationship MUST always override an inferred relationship if the name collides.
+
+Looking at the example below, a REST exporter would add the link `greeter` pointing to `/greeting/{userid}` on a 
+response returned from `/user/:userId`. As the value of the `userId` `PathParameter` is known, it is replaced with
+the actual value. `@PathParameter`s for which no predefined value can be inferred should be formatted according to
+[RFC 6570](https://tools.ietf.org/html/rfc6570) (as well as potential query parameters of the destination). 
+
+```swift
+var content: some Component {
+  Group("user", $userId.relationship("greeter", to: "greeting", $userId)) {
+    Handler()
+  }
+  Group("greeting", $userId) {
+    Greeter()
+  }
+}
+```
+
+### 2.3. Excluding relationships
+
+It may also be a sensible decision for a user to hide a inferred relationship. To support that
+a modifier can be introduce.
+
+Even if a user supplies a `.relationshipName` modifier, the relationship will still be hidden when `.hideRelationship` is used.
+
+Considering the example below, a request to `/` will not include a relationship pointing to `/test`.
+
+```swift
+var content: some Component {
+  Group("test".hideRelationship()) {
+    Handler()
+  }
+}
+```
+
+## 3. Using type information
+
+Checking the structure of the Webservice is already pretty powerful, but doesn't allow for relationships between handlers
+which are on the same level or on different paths of the path component tree.  
+This chapter highlights how we can nonetheless retrieve relationship information by looking at type information.
+
+### 3.1 Looking at the return type of `handle()`
+
+For this chapter we will look at the following example web service:
+
+```swift
+struct User: Identifiable {
+  var id: String
+  var name: String
+  var lastname: String
+  var email: String
+}
+
+struct TestService: WebService, Component {
+  @PathParameter var userId: User.ID
+
+  var content: some Component {
+    Group("user", $postId) {
+      UserHandler(userId: $userId)
+    }
+  }
+}
+```
+
+It can be observed that the handle function of the `UserHandler` returns a `User` instance.
+User conforms to the protocol `Identifiable`, so it is also known that every `User` instance is uniquely identified by the
+`id` property.  
+Furthermore the `UserHandler` is registered with a path parameter having the same `User.ID` type, meaning
+we can assume that the `/user/:userId` route can be used to query a certain `User` instance
+(meaning `:userId` translates to `User.id` property).
+
+_Note: As the type of a `@PathParameter` is a `LosslessStringConvertible` it most certainly is either a 
+`String` or an `Int`. Meaning checking if the type of the `User.id` property is equal to `User.ID` type might not be enough.
+Thus it could be necessary for `@PathParemeter`s to specify the type: e.g. `@PathParameter(type: User.self)` where
```

Comment:
Yes, this could be a needed extension. I am not sure if we would be able to use the Runtime framework to actually get to type defined by the developer of the user or it it is already mapped to `String` or `Int` when the program is executed. Otherwise explicitly adding this in the `@PathParameter` would be a possible option. We would also need to to the same of the `.http(.path)` option for `@Parameter`s.